internal/types/array.go

```diff
@@ -83,6 +84,10 @@ func (a *Array) GetByPath(path Path) (any, error) {
 
 // Set sets the value at the given index.
 func (a *Array) Set(index int, value any) error {
+	if a.frozen {
+		panic("this array is not mutable its been freezed")
+	}
```

Comment:
I think we should extract that info helper method:

```go
func (a *Array) checkFrozen() {
  if a.frozen {
    panic("array is frozen")
  }
}

func (a *Array) Set(index int, value any) error {
  a.checkFrozen()
  // ...
}
```

And then do the same for Document